mathics/builtin/base.py

```diff
@@ -456,20 +480,13 @@ def __new__(cls, *args, **kwargs):
     def init(self, *args, **kwargs):
         pass
 
-    @property
-    def is_literal(self) -> bool:
-        """
-        True if the value can't change, i.e. a value is set and it does not
-        depend on definition bindings. That is why, in contrast to
-        `is_uncertain_final_definitions()` we don't need a `definitions`
-        parameter.
-        """
-        # FIXME: figure out what the right thing to do here is.
-        # For now we will be pessimistic.
-        return False
-
 
 class AtomBuiltin(Builtin):
```

Comment:
Maybe this should be a subclass of ``InstanceableBuiltin`` (or whatever be a better name for it), or vice-versa. Now, I think this is confusing...

Reply:
`InstranceableBuiltin` or `SharedImplmenetationofBuiltinExceptWhenItIsn't` is some implemention of a Builtin. The "FIXME" says that it is a pessimistic guess because we have no idea. And as the FIXME suggests, this is bad. 

Probably instead inside Builtin where this has been moved, we should raise a RuntimeError because each real subclass should do the right thing. Notice that InstanceableBuiltin is _also_ a base class. If you create an object from that directly and start using it (rather than a subclass of that) then you are probably doing something wrong.

Reply:
I was not talking about the movement of the ``is_literal``, but about ``AtomBuiltin``which is something that we have had in the code for a while. As you were talking about re-examining this part of the class hierarchy, thought it was useful to say that this class still also confuses me.

Reply:
@mmatera It confuses me too. I suspect down the line it will be clear how to consolidate classes. Right now I was just being cautious. 

The distinction between Builtin and Atom and why Builtin is not a subclass of Atom feels weird, except when you drill into the details we find that Atom is very heavyweight. And Builtin adds the evaluation convention stuff (how functions are matched, using the method and docstring properties) which is also heavyweight. So combining these probably doesn't make sense. But it also doesn't feel well organizied.